docs/src/man/metadata.md

```diff
@@ -0,0 +1,206 @@
+
+julia> hasmetadata(df)
+
+julia> metadata(df)
+
+julia> empty!(df_meta)
+
+julia> hasmetadata(df)
+
+julia> metadata(df)
+
+julia> hasmetadata(df, :rating)
+
+julia> rating_meta = metadata(df, :rating)
+
+julia> rating_meta["name"] = "First and last name of a player"
+
+julia> rating_meta["rating"] = "ELO rating in classical time control"
+
+julia> rating_meta["date"] = "Rating date in yyyy-u format for \"english\" locale"
+
+julia> hasmetadata(df, :rating)
+
+julia> metadata(df, :rating)
+
+julia> metadata.(Ref(df), names(df))
+
+julia> empty!(rating_meta)
+
+julia> hasmetadata(df, :rating)
+
+julia> metadata(df, :rating)
+```
+
+As a practical tip if you have metadata attached to some object
+(either data frame or data frame column) and you want to propagate it to
+some new object you can use:
+* `copy!` to fully overwrite destination object metadata with source object
+  metadata;
+* `merge!` to add destination object metadata with source object metadata
+  (overwriting duplicates).
+
+## Propagation of metadata
+
+An important design feature of metatada is how it is handled when you transform
+data frames.
+
+!!! note
+
+    The provided rules might change in the future. Any change to metadata
+    propagation rules will not be considered to be a breaking change
+    in DataFrames.jl and can be done in any minor release of DataFrames.jl.
+    Such changes might be made based on users' feedback about what metadata
+    propagation rules are most convenient in practice.
+
+The general design rules for propagation of table and column level metadata
+are as follows:
+* if some object (data frame or column) is mutated in-place its metadata
+  is not changed.
+* preferably (this is not guaranteed in all cases as sometimes it might be hard
+  to detect that this condition is met) if some object is moved as a whole
+  (either by coping or by aliasing) then its metadata is propagated.
+* in all other cases if a new object is allocated from an existing object and
+  this transformation is not a copy or aliasing then metadata is dropped.
+
+The rationale behind these rules is that metadata can be arbitrary information
+about the object and any change of the object that the metadata is attached to
+might invalidate it.
+
+The concrete rules described below are derived from the general principles
+explained above by DataFrames.jl.
+
+TODO: the list above is not finished yet, but show the intention
+
+### Operations that preserve table-level metadata
+
+* `dropmissing!`
+* `filter!`
+* `insertcols!`
+* `invpermute!`
+* `permute!`
+* `reverse!`
+* `setindex!`
+* `shuffle!`
+* `unique!`
+
+### Operations that propagate table-level metadata
+
+* `copy`
+* `getindex` with `!` or `:` as row selector and `:` as column selector
+* `DataFrame`
+
+### Operations that preserve column-level metadata
+
+* `select!` and `transform!` if a single operation is passed selector
+  that selects one or multiple columns
```

Comment:
I don't really get what this means. What I would have imagined is that these would keep metadata when selecting a column without applying any transformation to it. This would notably include all columns that are kept (because not mentioned) by `transform!`.

Reply:
This rule will be gone